src/auth/src/credentials/idtoken.rs

```diff
@@ -129,33 +131,167 @@ fn instant_from_epoch_seconds(secs: u64) -> Option<Instant> {
     }
 }
 
+/// Verifier is used to verify OIDC ID Tokens.
+#[derive(Debug)]
+pub(crate) struct Verifier {
+    jwk_client: JwkClient,
+    audience: Option<String>,
+    email: Option<String>,
+    jwks_url: Option<String>,
+    clock_skew: Duration,
+}
+
+impl Default for Verifier {
+    fn default() -> Self {
+        Self {
+            jwk_client: JwkClient::new(),
+            audience: None,
```

Comment:
I believe we should make audience check required. I see that other langs are keeping it optional but this is one of the most important claims in the id token and it needs to be validated to make sure that the service is not using a token that is not intended for it.